test/dialect/test_sqlite.py

```diff
@@ -1362,6 +1363,289 @@ def test_on_conflict_clause_primary_key_constraint(self):
+            stmt,
+            "CREATE TABLE dst AS SELECT 1 AS x, 'a' AS y FROM src",
+        )
+
+    def test_explicit_temp_schema_without_keyword(self):
+        # When not using temporary but schema is temp (any case), qualify
+        src = self._source()
+        stmt = CreateTableAs(
+            select(src.c.id).select_from(src),
+            "dst",
+            schema="TEMP",
+        )
+        self.assert_compile(
+            stmt,
+            'CREATE TABLE "TEMP".dst AS SELECT src.id FROM src',
+        )
+
+    def test_if_not_exists(self):
+        src = self._source()
+        stmt = CreateTableAs(
+            select(src.c.id, src.c.name).select_from(src),
+            "dst",
+            if_not_exists=True,
+        )
+        self.assert_compile(
+            stmt,
+            "CREATE TABLE IF NOT EXISTS dst AS "
+            "SELECT src.id, src.name FROM src",
+        )
+
+    def test_union_all_smoke(self):
+        # Proves CTAS wraps a UNION ALL and preserves compound ordering.
+        a, b = self._two_sources()
+        u = (
+            select(a.c.id)
+            .select_from(a)
+            .union_all(select(b.c.id).select_from(b))
+            .order_by("id")  # order-by on the compound
+            .limit(3)
+        )
+        stmt = CreateTableAs(u, "dst")
+        self.assert_compile(
+            stmt,
+            "CREATE TABLE dst AS "
+            "SELECT a.id FROM a UNION ALL SELECT b.id FROM b "
+            "ORDER BY id LIMIT 3 OFFSET 0",
+        )
+
+    def test_cte_smoke(self):
+        # Proves CTAS works with a WITH-CTE wrapper and labeled column.
+        a, _ = self._two_sources()
+        cte = select(a.c.id.label("aid")).select_from(a).cte("u")
+        stmt = CreateTableAs(select(cte.c.aid), "dst")
+        self.assert_compile(
+            stmt,
+            "CREATE TABLE dst AS "
+            "WITH u AS (SELECT a.id AS aid FROM a) "
+            "SELECT u.aid FROM u",
+        )
+
+    def test_union_all_with_inlined_literals_smoke(self):
+        # Proves literal_binds=True behavior applies across branches.
+        a, b = self._two_sources()
+        u = (
+            select(literal(1).label("x"))
+            .select_from(a)
+            .union_all(select(literal("b").label("x")).select_from(b))
+        )
+        stmt = CreateTableAs(u, "dst")
+        self.assert_compile(
+            stmt,
+            "CREATE TABLE dst AS "
+            "SELECT 1 AS x FROM a UNION ALL SELECT 'b' AS x FROM b",
+        )
+
+
+class CreateTableAsSQLiteBehavior(fixtures.TestBase):
+    __only_on__ = "sqlite"
+    __backend__ = True
+
+    @staticmethod
+    def _source():
+        return table("src", column("id"), column("name"))
+
+    def setup_test(self):
+        # define a physical src table and seed a couple rows
+        self.src_def = "CREATE TABLE src (id INTEGER PRIMARY KEY, name TEXT)"
+        self.seed_def = "INSERT INTO src (name) VALUES ('a'), ('b')"
+
+    def teardown_test(self):
```

Comment:
note in the `exec_sql()` function at the top of the file it has a comment "convert all tests to not use this" :)   so for these raw SQL fixtures, I'd do it like this:

```
def teardown_test(self):
    with testing.db.begin() as conn:
         conn.exec_driver_sql(...)
         conn.exec_driver_sql(...)
         conn.exec_driver_sql(...)
         ...

```

but also I think these drops can all be turned into fixtures, see my other comments